src/vulkan-renderer/wrapper/once_command_buffer.cpp

```diff
@@ -28,7 +28,9 @@ OnceCommandBuffer::OnceCommandBuffer(OnceCommandBuffer &&other) noexcept
       m_command_buffer_created(other.m_command_buffer_created) {}
 
 OnceCommandBuffer::~OnceCommandBuffer() {
-    m_command_buffer.reset();
+    if (m_command_buffer != nullptr) {
+        m_command_buffer.reset();
+    }
```

Comment:
I don't think this is necessary since `std::unique_ptr::reset` works even if the pointer is null (like `free`).

Reply:
Since the member variable won't have a longer lifetime than the parent?

Reply:
Not really, it's just because no matter the state of the unique pointer, calling `reset` always works. Also I guess you could say the member variable has a longer lifetime than the parent (since it gets moved), not sure the exact definition of lifetime in that case though.

Reply:
Sounds good.  Do you want me to take the Destroy calls out of the ifs too?  It is no big deal to change.

Reply:
Up to you :), I guess it doesn't really matter too much. Btw I will open an issue about having the logs in the destructors at all.

Reply:
Sounds good.

I pushed up changes to the unique pointer code; I left the Destroys in the ifs for now.  If we end up removing the log messages, we can probably just take out the ifs and the logs at once.